### PR TITLE
add conditional header split

### PR DESCRIPTION
**Pull request by DavZim**
This PR adds a first (discussion) version to #187.
It implements an unordered set in cpp which keeps track of all headers which can be split (I didnt find a comprehensive list nor extensive documentation for each header which lists if it can have multiple elements, instead I looked through multiple examples (see comment in code) and took the ones which have examples with multiple elements).

As mentioned in the issue, I found no easy way to test this properly, as `app$process_request()` does not call `parse_headers()`. Instead I used the manual approach:

```r
library(RestRserve)
app = Application$new()
app$add_get("/foo", FUN = function(.req, .res) {
  print("/foo")
  print(.req$headers)
  .res$set_body(42)
  .res$set_status_code(200)
})

backend = BackendRserve$new()
backend$start(app, http_port = 8080)
```

and then using CURL I test the following:

```bash
curl http://localhost:8080/foo -H "If-Modified-Since: Tue, 15 Mar 2022 10:27:07 GMT" -H "Accept: abc, def, ghi"
```

which results in the following output in the logs of the Rsession:
```
[1] "/foo"
$accept
[1] "abc" "def" "ghi"

$`if-modified-since`
[1] "Tue, 15 Mar 2022 10:27:07 GMT"

$`user-agent`
[1] "curl/7.79.1"

$host
[1] "localhost:8080"
```

As we can see, the `Accept` header is split, but the `If-Modified-Since` header is not split.

Let me know what you think about this!

### CODE REVIEW

**Comment by artemklevtsov** [comment]
Code is ok. Need tests.
I'm not sure if it's worth letting the user control this.

**Comment by DavZim** [comment]
Agree on the tests, do you know how we can test this?

**Comment by artemklevtsov** [comment]
Please look at here: https://github.com/rexyai/RestRserve/blob/master/inst/tinytest/test-parse-headers.R

**Comment by dselivanov** [comment]
I think it worth to make list of splittable headers public/ not hardcoded. So it is possible to modify it before backend starts.

**Comment by DavZim** [comment]
Any updates on this? Are there any further tasks that need to be done before we can merge this?
Thanks

**Comment by dselivanov** [comment]
@DavZim I've pushed some updates which move split header to standard R options. Also I'm thinking where to put documentation for such things. Suggestions?

**Comment by DavZim** [comment]
Do you mean long form of documentation such as a vignette or shorter: a "chapter" in the readme?

The functionality itself is documented in Backend, right?

**Comment by dselivanov** [comment]
Short paragraph. New "split headers" option in theory should hold same logic for all backends (if for example we add `httpuv`).

File: NEWS.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 ## Changelog
 
+* 2022-06-08 - 1.2.0
+    * Expose option to control which HTTP headers need to be split by comma during parsing. See `options("RestRserve.headers.split")`. See #187, #189. Thanks @DavZim.
+    * Improved ETaf Middleware - see #188. Thanks @DavZim.
```
Comment (artemklevtsov) [inline NEWS.md:5]: `s/Etaf/Etag`

**Pull request merged**